From c63fc060503c7481a6775cd030ca626379cdeb3f Mon Sep 17 00:00:00 2001
From: Muna Sinada <quic_msinada@quicinc.com>
Date: Mon, 20 Sep 2021 16:15:02 -0700
Subject: [PATCH 19/27] mac80211: Support disabling EHT mode

Allow user to disable EHT mode. This forces EHT capable interfaces
to disable during association.

Signed-off-by: Muna Sinada <quic_msinada@quicinc.com>
---
 net/mac80211/mlme.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -6936,7 +6936,7 @@ int ieee80211_mgd_assoc(struct ieee80211
 			conn_flags |= IEEE80211_CONN_DISABLE_HE;
 			conn_flags |= IEEE80211_CONN_DISABLE_EHT;
 			netdev_info(sdata->dev,
-				    "disabling HT/VHT/HE due to WEP/TKIP use\n");
+				    "disabling HT/VHT/HE/EHT due to WEP/TKIP use\n");
 		}
 	}
 
@@ -6951,7 +6951,7 @@ int ieee80211_mgd_assoc(struct ieee80211
 	}
 
 	if (req->flags & ASSOC_REQ_DISABLE_HT) {
-		mlme_dbg(sdata, "HT disabled by flag, disabling HT/VHT/HE\n");
+		mlme_dbg(sdata, "HT disabled by flag, disabling HT/VHT/HE/EHT\n");
 		conn_flags |= IEEE80211_CONN_DISABLE_HT;
 		conn_flags |= IEEE80211_CONN_DISABLE_VHT;
 		conn_flags |= IEEE80211_CONN_DISABLE_HE;
